windows/devices/uwb/UwbConnector.cxx

```diff
@@ -915,8 +939,90 @@ UwbConnector::CallbacksPresent()
 }
 
 void
-UwbConnector::DeregisterEventCallback(::uwb::RegisteredCallbackToken* token)
+UwbConnector::DeregisterEventCallback(std::weak_ptr<::uwb::RegisteredCallbackToken> token)
 {
-    // TODO implement
+    auto tok = token.lock();
+    if (not tok) {
+        return;
+    }
     std::lock_guard eventCallbacksLockExclusive{ m_eventCallbacksGate };
+
+    DeregisterEventCallback(*tok);
 }
+
+void
+UwbConnector::DeregisterEventCallback(::uwb::RegisteredCallbackToken token)
+{
+    auto callbackId = token.callbackId;
+    auto isDeviceEventCallback = token.isDeviceEventCallback;
+
+    // first remove the token from the list of tokens
+    for (auto it = std::cbegin(m_tokens); it != std::cend(m_tokens); it++) {

```

Comment:
You could use `std::erase_if` here instead. For example,

std::erase_if(m_tokens, [&callbackId](const auto& token) {
    return token->callbackId == callbackId;
});

Reply:
+1, std:;erase_if() is is safer. `erase()` on its own invalidates the iterator, so calling `it++` here would be accessing invalid memory. `erase()` returns the iterator following the erased element, so the idiomatic way to erase is to do this:

```
it = containerItCameFrom.erase(it);
```

Reply:
erase_if looks at all the elements so I don't think it's more efficient (although it does make for nice one-liner). I believe the way I implemented it will break the for loop immediately after erasing, so it never executes it++ on an invalid iterator.

Reply:
I totally missed the `break`. So, you're right on both points: `erase_if` is less efficient and the way you did it was 100% safe.

👍